salt/states/pem.py

```diff
@@ -0,0 +1,173 @@
+"""
+Provide easier operations on certificate files
+=====================================================================
+
+Main purpose for this is to provide easier and human readable info
+regarding changes if files contains certificates.
+
+Requirement is `cryptography` python package.
+
+.. code-block:: yaml
+
+/etc/ssl/postfix/mydomain/mydomain.pem:
+  pem.managed:
+    - source: salt://files/etc/ssl/mydomain.pem
+    - user: root
+    - group: postfix
+    - dir_mode: 711
+    - makedirs: True
+    - template: jinja
+"""
+
+import logging
+
+import salt.utils.files
+
+try:
+    from cryptography import x509
+
+    HAS_CRYPTOGRAPHY = True
+except ImportError:
+    HAS_CRYPTOGRAPHY = False
+
+__virtualname__ = "pem"
+
+
+def __virtual__():
+    if not HAS_CRYPTOGRAPHY:
+        return (False, "Could not load cryptography")
+    return __virtualname__
+
+
+log = logging.getLogger(__name__)
+
+
+def managed(
+    name,
+    source=None,
+    source_hash=None,
+    source_hash_name=None,
+    user=None,
+    group=None,
+    mode=None,
+    skip_verify=None,
+    defaults=None,
+    attrs=None,
+    context=None,
+    saltenv="base",
+    skip_conditions=False,
+    **kwargs,
+):
+    """
+    State manage certificates as files and provide minimal overview of
+    expiration and common name in comment.
+
+    It can handle one certificate or full privkey->cert->chain files.
+
+    Conditions can provide easy way how to match agains Common name,
+    or make sure that only newer certificates will be salted.
+    If conditions are not required use this as argument to state:
+    `skip_conditions=True`
+    Or set pillar in same manner on command line:
+    `salt-call state.apply some_other_state pillar="{skip_conditions: True}"`
```

Comment:
These should be under a list of Args including all args for the function.

```
Args:

    name (str): The full path to the certificate file on the target system

    source (str): The full path to the certificate source

    source_hash (str): The hash of the source file

    source_hash_name (str): ...
```

Or provide a link to the `file.managed` docs to state that it uses similar parameters, then document the additional parameters provided by this function.